src/merkle.rs

```diff
@@ -237,19 +549,109 @@ impl<T: Element, A: Algorithm<T>, K: Store<T>, U: Unsigned> MerkleTree<T, A, K,
         );
 
         Ok(MerkleTree {
-            data,
+            data: Some(data),
             leafs,
+            len: tree_len,
             height,
+            // FIXME: correct?
```

Comment:
still valid?

Reply:
Yes, the comments there need to be removed.

Reply:
Or rather, the FIXME is not valid